### PR TITLE
add github action recipe for creating deployment referencing storage and infra blocks

### PR DESCRIPTION
**Pull request by zzstoatzz**
## Description
Meant to leverage the separation of environment already provided by github repos, this workflow could be used by any prefect user that needs to create dockerized deployments - can work with any container registry and on any repo branch as long as GitHub secrets and corresponding prefect blocks/work-queues exist.

Defining the `{my}-deployment.yaml` and `Dockerfile` explicitly as "inputs" show that we can use this for any chosen repo structure (monorepos, depoyment 1-to-1 with repo, etc)

## New Recipe Checklist

- [x] My PR is in the format of `Add <project-name> recipe`
- [x] My recipe is reproducible and explains everything needed to run successfully. If my code has external dependencies, I make mention of them.
- [x] My code is easily understandable and/or well-commented.

### CODE REVIEW

File: devops/github-actions/general-docker-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        run: pip install -U wheel pip prefect s3fs
+
+      - name: Prefect Cloud login
+        run: |
+          prefect config set PREFECT_API_KEY=$API_KEY
+          prefect config set PREFECT_API_URL=$API_URL
+
+        env:
+          API_KEY: ${{ secrets.PREFECT_API_KEY }}
+          API_URL: ${{ secrets.PREFECT_API_URL }}
```
Comment (EmilRex) [inline devops/github-actions/general-docker-deploy.yaml:48]: I believe that if you just set the environment variables (`PREFECT_API_KEY` and `PREFECT_API_URL`)  you don't have to run `prefect config set`. So you should be able to get rid of this step and simply pass the variables in the next task (if you want).

Comment (zzstoatzz) [reply devops/github-actions/general-docker-deploy.yaml:48]: I was able to get it to work this way, but then you'd have to pass those secrets into `env` for each prefect command 
```
      - name: Build Prefect deployment
        run: prefect deployment build ./prefect/flows/hello.py:$FLOWNAME -n $DEPLOYMENT_NAME -t $TAG -ib $INFRA -sb $STORAGE -o $DEPLOYMENTYAML -q $WORK_QUEUE
        env:
          DEPLOYMENT_NAME: hello-marvin
          TAG: dev
          INFRA: kubernetes-job/default-k8s-job
          STORAGE: s3/flow-script-storage-${{ steps.extract_branch.outputs.branch }}
          WORK_QUEUE: "main"
          PREFECT_API_KEY: ${{ secrets.PREFECT_API_KEY }}
          PREFECT_API_URL: ${{ secrets.PREFECT_API_URL }}
          
      - name: Apply Prefect deployment
        run: prefect deployment apply $DEPLOYMENTYAML
        env:
          PREFECT_API_KEY: ${{ secrets.PREFECT_API_KEY }}
          PREFECT_API_URL: ${{ secrets.PREFECT_API_URL }}
```

imo it seems easier to do it once up front have all subsequent prefect commands just use the config setting like
```
      - name: Prefect Cloud login
        run: |
          prefect config set PREFECT_API_KEY=${{ secrets.PREFECT_API_KEY }}
          prefect config set PREFECT_API_URL=${{ secrets.PREFECT_API_URL }}

      - name: Build Prefect deployment
        run: prefect deployment build ./prefect/flows/hello.py:$FLOWNAME -n $DEPLOYMENT_NAME -t $TAG -ib $INFRA -sb $STORAGE -o $DEPLOYMENTYAML -q $WORK_QUEUE
        env:
          DEPLOYMENT_NAME: hello-marvin
          TAG: dev
          INFRA: kubernetes-job/default-k8s-job
          STORAGE: s3/flow-script-storage-${{ steps.extract_branch.outputs.branch }}
          WORK_QUEUE: "main"
          
      - name: Apply Prefect deployment
        run: prefect deployment apply $DEPLOYMENTYAML
        
```

Comment (gabcoyne) [reply devops/github-actions/general-docker-deploy.yaml]: 👍🏻

**Review by gabcoyne — APPROVED** [review]
Looks good

**Review by gabcoyne — APPROVED** [review]

File: devops/github-actions/general-docker-deploy.yaml (hunk 1 above)
Comment (zzstoatzz) [reply devops/github-actions/general-docker-deploy.yaml]: ahh that's true, I'll do that

**Review by serinamarie — APPROVED** [review]
Looks great, thanks for adding this!

**Pull request merged**